containers/payments/subscription/BlackFridayModal.js

```diff
@@ -0,0 +1,190 @@
+            [YEARLY]: c('Title').jt`Billed as ${amount} (${notice})`,
+            [TWO_YEARS]: c('Title').jt`Billed as ${amount} (${notice})`
+        }[cycle]);
+
+    const AFTER_INFO = ({ cycle, amount, notice }) =>
+        ({
+            [MONTHLY]: c('Title')
+                .jt`(${notice}) After one month, your subscription will automatically renew at the regular price of ${amount} every month.`,
+            [YEARLY]: c('Title')
+                .jt`(${notice}) After one year, your subscription will automatically renew at the regular price of ${amount} every year.`,
+            [TWO_YEARS]: c('Title')
+                .jt`(${notice}) After two years, your subscription will automatically renew at the regular price of ${amount} every two years.`
+        }[cycle]);
+
+    const getBundlePrices = async () => {
+        const result = await Promise.all(
+            bundles.map(({ planIDs = [], cycle = DEFAULT_CYCLE, couponCode }) => {
+                return Promise.all([
+                    api(
+                        checkSubscription({
+                            PlanIDs: planIDs,
+                            CouponCode: couponCode,
+                            Currency: currency,
+                            Cycle: cycle
+                        })
+                    ),
+                    api(
+                        checkSubscription({
+                            PlanIDs: planIDs,
+                            Currency: currency,
+                            Cycle: MONTHLY
+                        })
+                    )
+                ]);
+            })
+        );
+
+        updatePricing(
+            result.reduce((acc, [withCoupon, withoutCouponMonthly], index) => {
+                acc[index] = {
+                    withCoupon: withCoupon.Amount + withCoupon.CouponDiscount,
+                    withoutCoupon: withCoupon.Amount,
+                    withoutCouponMonthly: withoutCouponMonthly.Amount
+                };
+                return acc;
+            }, {})
+        );
+    };
+
+    useEffect(() => {
+        withLoading(getBundlePrices());
+    }, []);
+
+    return (
+        <FormModal title={c('Title').t`Black Friday sale`} loading={loading} footer={null} {...rest}>
+            {loading ? (
+                <Loader />
+            ) : (
+                <>
+                    <Alert>{c('Info').t`Don't miss out on limited time discounts for newcomers!`}</Alert>
+                    <div className="flex flex-nowrap flex-spacebetween">
+                        {bundles.map(({ name, cycle, planIDs, popular, couponCode, pourcentage }, index) => {
+                            const key = `${index}`;
+                            const { withCoupon = 0, withoutCouponMonthly = 0 } = pricing[index] || {};
+                            const monthlyPrice = (
+                                <Price currency={currency} suffix="/mo">
+                                    {withCoupon / cycle}
```

Comment:
Is `withCoupon / cycle` guaranteed to give a number which makes sense as a price? E.g. `15.3333333` wouldn't be nice

Reply:
Yes.